File: backend/src/Models/user.model.js
import mongoose, { mongo } from "mongoose";
const UserSchema = new mongoose.Schema(
  {
    name: {
      type: String,
      required: true,
    },
    email: {
      type: String,
      required: true,
      unique: true,
    },
    mobilenumber: {
      type: String,
      required: true,
      unique: true,
    },
    avatar: {
      type: String,
      default: "default-avatar-url.jpg",
    },
    password: {
      type: String,
      required: true,
      unique: true,
    },
    role: {
      type: String,
      enum: ["SUPER_ADMIN", "MINE_ADMIN", "SUPERVISOR"],
      default: "SUPERVISOR",
    },
    mineid: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Mine",
    },
  },
  {
    timestamps: true,
  },
);

export const UserModel = mongoose.model("User", UserSchema);

File: frontend/lib/data/models/auth/user_model.dart
import 'package:mine_wadhwani/domain/entities/auth/user_entity.dart';

class UserModel extends UserEntity {
  const UserModel({
    required super.id,
    required super.name,
    required super.email,
    required super.mobilenumber,
    required super.avatar,
    required super.role,
    super.mineid,
    required super.token,
  });

  factory UserModel.fromJson(Map<String, dynamic> json, {String? token}) {
    return UserModel(
      id: json['_id'] as String,
      name: json['name'] as String,
      email: json['email'] as String,
      mobilenumber: json['mobilenumber'] as String,
      avatar: json['avatar'] as String? ?? 'default-avatar-url.jpg',
      role: json['role'] as String? ?? '',
      mineid: json['mineid'] as String?,
      token: token ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'name': name,
      'email': email,
      'mobilenumber': mobilenumber,
      'avatar': avatar,
      'role': role,
      'mineid': mineid,
      'token': token,
    };
  }
}
